src/mixins/date-mixin.ts

```diff
@@ -243,51 +281,58 @@ export default class DateMixin extends Vue {
   }
 
   /**
-   * Returns the earliest of two simple date strings (YYYY-MM-DD).
+   * Returns the earliest of two date strings (YYYY-MM-DD).
    * @param date1 first date
    * @param date2 second date
    */
-  earliestDate (date1: string, date2: string): string {
+  earliestYyyyMmDd (date1: string, date2: string): string {
     // safety check
     if (!date1 && !date2) return null
 
     // edge cases
     if (!date1) return date2
     if (!date2) return date1
 
-    return (this.compareDates(date1, date2, '<') ? date1 : date2)
+    return (this.compareYyyyMmDd(date1, date2, '<') ? date1 : date2)
   }
 
   /**
-   * Returns the latest of two simple date strings (YYYY-MM-DD).
+   * Returns the latest of two date strings (YYYY-MM-DD).
    * @param date1 first date
    * @param date2 second date
    */
-  latestDate (date1: string, date2: string): string {
+  latestYyyyMmDd (date1: string, date2: string): string {
     // safety check
     if (!date1 && !date2) return null
 
     // edge cases
     if (!date1) return date2
     if (!date2) return date1
 
-    return (this.compareDates(date1, date2, '>') ? date1 : date2)
+    return (this.compareYyyyMmDd(date1, date2, '>') ? date1 : date2)
   }
 
   /**
-   * The number of days that 'date' is from today.
+   * Returns the number of days that 'date' is from today in Pacific timezone.
    * @returns -1 for yesterday
    * @returns 0 for today
    * @returns +1 for tomorrow
    * @returns NaN in case of error
    */
-  daysFromToday (date: string): number {
+  daysFromToday (date: Date): number {
     // safety check
-    if (!date) return NaN
+    if (!isDate(date) || isNaN(date.getTime())) return NaN
+
+    // set "date" to 12:00 am Pacific
+    date.setHours(0, 0, 0, 0)
+
+    // compute "today" at 12:00 am Pacific
+    const today = this.getCurrentJsDate
+    today.setHours(0, 0, 0, 0)
```

Comment:
Internally `getCurrentJsDate` is stored in UTC with the current Pacific tz offset.
But when we call `setHours()`, it's done in local time.
So this works as expected.

Example:
![image](https://user-images.githubusercontent.com/10002889/143334103-d068711e-3f83-4e15-96d1-6d5bffacfe52.png)